src/pal/src/misc/miscpalapi.cpp

```diff
@@ -267,7 +267,7 @@ PAL_Random(
             DWORD offset = 0;
             do
             {
-                DWORD n = read(rand_des, (BYTE*)lpBuffer + offset , dwLength - offset);
+                auto n = read(rand_des, (BYTE*)lpBuffer + offset , dwLength - offset);
```

Comment:
We avoid `auto` unless it is obvious what the type is. `size_t` ?

Reply:
It was `ssize_t` and since we are compiling with `-Wall`, it is hard to know if all supported platforms share the same. We could cast RHS to `DWORD` instead.

Reply:
Actually `DWORD` is `uint32_t`, so the comparison on next line with `-1` causes error with GCC. Trying with `ssize_t` on `LHS`.